extension/parquet/include/parquet_file_metadata_cache.hpp

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// parquet_file_metadata_cache.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "duckdb/storage/object_cache.hpp" // ObjectCache
+#include "parquet_types.h"                 // parquet::format::FileMetaData
+#include <sys/stat.h>                      // time_t
```

Comment:
stat.h is not a C or C++ standard won't work on Windows, please instead add methods to the FileSystem class that handle this in an OS-specific manner there, so no other library needs to know about it